src/ControlSystem/Averager.cpp

```diff
@@ -158,35 +159,69 @@ double Averager<DerivOrder>::average_time(const double time) const noexcept {
 }
 
 template <size_t DerivOrder>
-std::array<DataVector, DerivOrder + 1> Averager<DerivOrder>::get_derivs() const
-    noexcept {
-  static_assert(DerivOrder == 2,
-                "Finite differencing currently only implemented for DerivOrder "
-                "= 2. If a different DerivOrder is desired, please add stencil "
-                "coefficients for that order.");
-
+std::array<DataVector, DerivOrder + 1> Averager<DerivOrder>::get_derivs()
+    const noexcept {
   // initialize the finite difference coefs
-  std::array<std::array<double, 3>, 3> coefs{};
-
-  const double delta_t1 = times_[0] - times_[1];
-  const double delta_t2 = times_[1] - times_[2];
-  const double delta_t = delta_t1 + delta_t2;
-  const double denom = 1.0 / delta_t;
-  const double dts = delta_t1 * delta_t2;
-
-  // set coefs for function value
-  coefs[0] = {{1.0, 0.0, 0.0}};
-  // first deriv coefs
-  coefs[1] = {{(2.0 + delta_t2 / delta_t1) * denom, -delta_t / dts,
-               delta_t1 / delta_t2 * denom}};
-  // second deriv coefs
-  coefs[2] = {{2.0 / delta_t1 * denom, -2.0 / dts, 2.0 / delta_t2 * denom}};
-
-  std::array<DataVector, 3> result =
-      make_array<3>(DataVector(raw_qs_[0].size(), 0.0));
+  std::array<std::array<double, DerivOrder + 1>, DerivOrder + 1> coefs{};
```

Comment:
Please add a reference to where the formulae for the coefs come from.  I verified they are correct by comparing to a numerical implementation of Fornberg's algorithm (https://doi.org/10.1090/S0025-5718-1988-0935077-0).

Reply:
So I got these formulae from SpEC (specifically the FiniteDifference1d.cpp file). I looked there, but I couldn't find a source for where they got the formulae either. Should I just confirm with this paper as well and use that as my reference?

Reply:
@markscheel is there a good reference for these stencils?

Reply:
I talked with @markscheel and he said he just derived them himself.

Reply:
okay, I think it is sufficient just to comment that these are the weights of the Lagrange interpolating polynomial and it derivatives evaluated at `times_[0]`